### ISSUE TITLE
Astro transform fails on several forms of multiline exports

### ISSUE THREAD

**Issue by duanwilliam** [withastro/compiler#554]

**Comment by delucis** [comment]
Additional case reported [on StackOverflow](https://stackoverflow.com/questions/74588321/how-to-fix-the-unexpected-while-creating-a-custom-react-astro-component), which provides [a CodeSandbox reproduction](https://codesandbox.io/p/sandbox/loving-kare-m8ggi7).

#### Works

```astro
---
import { SomeType } from 'package';

export type Props = SomeType & {
  margin?: number | string
};
---
```

#### Doesn’t work

```astro
---
import { SomeType } from 'package';

export type Props = {
  margin?: number | string
} & SomeType;
---
```

Throws an error:

> Transform failed with 1 error: ERROR: Unexpected "&"

**Comment by TymekDev** [comment]
I see this issue marked as `p2-has-workaround`. What is this workaround? I looked for making Astro LSP ignore one particular line, but haven't found anything of use. Is there a way other than turning off LSP before write?

**Comment by Princesseuh** [comment]
Just a note to anyone visiting this issue that exporting `Props` is not needed (and never has been) unless you're actually using it in other files.

**Comment by yanickrochon** [comment]
Prettier formats code with this syntax, and it works fine in Next.js (haven't tried in other projects), so why is this failing, here?

Is unions like

```tsx
type Foo =
 | string
 | number;
```
still valid TS (the documentation page no longer show this format).

**Comment by Princesseuh** [comment]
> Prettier formats code with this syntax, and it works fine in Next.js (haven't tried in other projects), so why is this failing, here?
> 
> Are unions like
> 
> ```tsx
> type Foo =
>  | string
>  | number;
> ```
> still valid TS (the documentation page no longer show this format).

Yes, this is valid TS. It's a bug in our compiler, only affects exported things though

**Comment by xavdid** [comment]
I was able to work around this by defining the type separately from the function:

```ts
type StaticPathResult = () => Promise<
  { params: { year: number }; props: Props }[]
>;

export const getStaticPaths: StaticPathResult = async () => {
  // ...
}
```

**Comment by fratzinger** [comment]
In my case it was prettier auto formatting forcing me into this issue.

My current workaround is to transform this:
```ts
type Test =
  | 'this'
  | 'test':
```
to this:
```ts
// prettier-ignore
type Test = 'this' | 'test'
```

**Comment by gersomvg** [comment]
Just hit this issue for the first time. I can confirm that altering the line breaks somewhat combined with a `// prettier-ignore` solves the issue.

**Comment by joediton** [comment]
I've hit this issue. Removing export resolves this issue

```javascript
export type TBlock =
  | TBlock_Text
  | TBlock_CardsList
  | TBlock_CtaSection
  | TBlock_Iframe
  | TBlock_Hero;
```

My work arround
```javascript
type TBlock_Union =
  | TBlock_Text
  | TBlock_CardsList
  | TBlock_CtaSection
  | TBlock_Iframe
  | TBlock_Hero;

export TBlock = TBlock_Union
```

**Comment by Princesseuh** [comment]
Hello, the Go compiler for Astro is no longer maintained. All future development will be focused on [our new Rust compiler](https://github.com/withastro/compiler-rs). If this issue is still relevant and you can reproduce it on the Rust compiler, please open a new issue there. Thank you!
